File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    arr = []
    for n in number:
        n = int(n)
        while k > 0 and len(arr) > 0:
            if arr[-1] >= n:
                break
            arr.pop()
            k -= 1
        arr.append(n)
    if (k > 0):
        arr = arr[0:len(arr) - k]
    return "".join(map(str, arr))

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    let arr = [];
    for (let i = 0; i < number.length; i++) {
        let num = number[i] * 1;
        while (arr.length && k > 0) {
            if (arr[arr.length - 1] >= num) break;
            arr.pop();
            k--;
        }
        arr.push(num);
    }
    if (k > 0) arr = arr.slice(0, arr.length - k);
    return arr.join('');
}